src/lib/builders/menubar/create.ts

```diff
@@ -422,11 +444,35 @@ export function createMenubar(props?: CreateMenubarProps) {
 		};
 	});
 
+	const unsubs: Array<() => void> = [];
+
+	effect([activeMenu], ([$activeMenu]) => {
+		if (!isBrowser) return;
+
+		if ($activeMenu && !scrollRemoved) {
+			unsubs.push(removeScroll());
+			scrollRemoved = true;
+		}
+
+		// do something with remove scroll
+
+		return () => {
+			if (!get(activeMenu)) {
+				unsubs.forEach((unsub) => unsub());
+				scrollRemoved = false;
+			}
+		};
```

Comment:
```suggestion
	effect([activeMenu], ([$activeMenu]) => {
		if (!isBrowser) return;
				
		if (!$activeMenu) {
			unsubs.forEach((unsub) => unsub());
			scrollRemoved = false;
		} else if (!scrollRemoved) {
			unsubs.push(removeScroll());
			scrollRemoved = true;
		}
	)};
```

Wouldn't this achieve the same result?